src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationSource.cs

```diff
@@ -32,6 +33,11 @@ public abstract class FileConfigurationSource : IConfigurationSource
         public bool ReloadOnChange { get; set; }
 
         /// <summary>
+        /// Will be called if an uncaught exception occurs in FileConfigurationProvider.Load.
+        /// </summary>
+        public Action<Exception> OnLoadError { get; set; }
```

Comment:
We could consider making this a Func that returns bool to control whether we want to rethrow the exception.  That might be a nice quality of life improvement, but not sure how useful it is always...